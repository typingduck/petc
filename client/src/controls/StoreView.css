
#petc-storeview {
  border: 1px dashed #AAA;
  left: 70%;
  position: absolute;
  top: 50%;
  z-index: 0;
}
