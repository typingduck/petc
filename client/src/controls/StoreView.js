import React from 'react'
import './StoreView.css'

/**
 * A useful development tool for viewing the redux state in the browser.
 * Hidden during normal viewing.
 */
export default function StoreView (props) {
  if (props.controls.storeview) {
    return (
      <pre id='petc-storeview'>
        {props.docId}<br />
        {JSON.stringify(props.controls, null, 2)}
        <hr />
        {JSON.stringify(props.doc, null, 2)}
      </pre>
    )
  } else {
    return null
  }
}
