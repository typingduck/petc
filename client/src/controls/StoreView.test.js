/* global expect it */
import React from 'react'
import Renderer from 'react-test-renderer'

import StoreView from './StoreView'

it('renders storeview', () => {
  const props = {
    controls: { storeview: true },
    docs: { a: 1 }
  }
  const component = Renderer.create(<StoreView {...props}/>)
  expect(component.toJSON()).toMatchSnapshot()
})

it('hides storeview when not debugging', () => {
  const props = {
    controls: { storeview: false },
    docs: { a: 1 }
  }
  const component = Renderer.create(<StoreView {...props}/>)
  expect(component.toJSON()).toBeNull()
})
