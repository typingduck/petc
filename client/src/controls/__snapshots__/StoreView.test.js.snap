// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders storeview 1`] = `
<pre
  id="petc-storeview"
>
  {
    "storeview": true
  }
  <hr />
</pre>
`;
